examples/recording_service/leveldb-integration/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,180 @@
+#
+# (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+#
+#  RTI grants Licensee a license to use, modify, compile, and create derivative
+#  works of the Software.  Licensee has the right to distribute object form
+#  only for use with RTI products.  The Software is provided "as is", with no
+#  warranty of any type, including any warranty for fitness for any purpose.
+#  RTI is under no obligation to maintain or support the Software.  RTI shall
+#  not be liable for any incidental or consequential damages arising out of the
+#  use or inability to use the software.
+#
+cmake_minimum_required(VERSION 3.11)
+project (LevelDbIntegration)
+
+set(CMAKE_DEBUG_POSTFIX "d")
+
+# Find RTI Connext dependencies
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CONNEXTDDS_DIR}/resource/cmake")
+
+find_package(
+    RTIConnextDDS "6.0.1"
+    REQUIRED
+    COMPONENTS
+        core
+        routing_service
+)
+
+set(LEVELDB_LIB_NAME "leveldb")
+
+find_library(
```

Comment:
I downloaded and installed `LevelDB` and it has integration with CMake.
The way to activate it is adding the folder `<where LevelDB is installed>/lib64/cmake/leveldb/` to `CMAKE_MODULE_PATH` and you will be able to find it using `find_package`.


So the problem now: how can we find the folder where LevelDB is installed? Well, we can ask the user to provide it using the `LEVELDB_DIR` CMake variable but we can try to add some logic to try to find it... Like the default path where it is installed in Windows, Linux and Darwin.
Also, since we know the library will be installed under `lib64`, we can run `find_library` to find the library (without specifying any path to ask to CMake to search on its default paths) and them add `<where the library is placed>/cmake/leveldb/` to `CMAKE_MODULE_PATH` to be able to use `find_package`. Why can this be useful? If, in the future, they decide to add compiler flags, compiler definitions, dependencies with other libraries or other things, that information will be included in their CMake scripts and our `CMakeLists.txt` file will still be valid.

Reply:
Thank you Israel for the thorough review of the build file.

In a separate note, I can anticipate people will do the same things differently not followiing our methods. We should think about a creating some sort of template (e..g, a velocity one to be used with rtiddsgen) so that all of these things can be reused. We could also have a simple tool added at the top-level of the repository to generate the CMake.

Just leaving the thought here. We can discuss further how to do it if you agree

Reply:
One note about LevelDB: while its CMakeLists.txt is ready to build as shared library (we need that), the bundle doesn't provide that by default, if I recall correctly.
@iblancasa , when you said you installed levelDB, you mean you used an installation bundle or just unzipped something from github?

Reply:
@juanlu-rti I downloaded the source code, built it and installed the result. Also, I tried with the package from the Fedora repositories.